.github/workflows/build_portable_linux_artifacts.yml

```diff
@@ -118,39 +125,64 @@ jobs:
           amdgpu_families: ${{ inputs.amdgpu_families }}
           package_version: ${{ inputs.package_version }}
           extra_cmake_options: ${{ inputs.extra_cmake_options }}
-          BUILD_DIR: build
         run: |
           python3 build_tools/github_actions/build_configure.py --manylinux
 
+      - name: Start memory monitoring
+        if: ${{ inputs.monitor_memory }}
+        id: memory-monitor
+        shell: bash
+        continue-on-error: true
+        env:
+          BUILD_DIR: ${{ env.BUILD_DIR }}
+          JOB_NAME: ${{ github.job }}
+          PHASE: "Build Phase"
+        run: |
+          # Run the memory monitor start script and capture PID
+          MONITOR_PID=$(bash build_tools/github_actions/start_memory_monitor.sh)
+          echo "monitor-pid=$MONITOR_PID" >> $GITHUB_OUTPUT
+
       - name: Build therock-archives and therock-dist
         run: |
-          cmake --build build --target therock-archives therock-dist -- -k 0
+          cmake --build ${{ env.BUILD_DIR }} --target therock-archives therock-dist -- -k 0
+
+      - name: Stop memory monitoring
+        if: ${{ always() && inputs.monitor_memory && steps.memory-monitor.outputs.monitor-pid }}
+        shell: bash
+        continue-on-error: true
+        env:
+          MONITOR_PID: ${{ steps.memory-monitor.outputs.monitor-pid }}
+          BUILD_DIR: ${{ env.BUILD_DIR }}
+          JOB_NAME: ${{ github.job }}
+        run: |
+          # Run the memory monitor stop script
+          bash build_tools/github_actions/stop_memory_monitor.sh
```

Comment:
Same comments here:

* `shell: bash` is not needed
* `continue-on-error` may not be needed
* `env.BUILD_DIR` forwarding is not needed
* The comment doesn't add anything